### DISCUSSION TITLE
Set expiry time for key in garnet client

### DISCUSSION THREAD

**Discussion by VartuSharma** [microsoft/garnet#445] [Q&A] [answered]
Is there any way I can set expiration of key while setting the value using garnet.client?

**Comment by KevinHuebner** [comment]
No its not possible at the time, but if you use StackExchange.Redis as the Service for the Garnet Client (Server), you can use its method wirh StringSetAsync(key, value, TimeSpan.FromMinutes()) to set there an expiration time.

**Comment by Collaborator07** [comment, marked as answer]
> Is there any way I can set expiration of key while setting the value using garnet.client?

await db.ExecuteForStringResultAsync("EXPIRE", [key, "30", "NX"]);
